tests/cross/src/test/scala/tests/pc/InsertInferredMethodSuite.scala

```diff
@@ -0,0 +1,527 @@
+       |      val path = Files.createTempDirectory("")
+       |      def otherMethod(arg0: Path): String = ???
+       |      method1(otherMethod(path))
+       |    }
+       |}
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : String => Int) = 123
+       |    method1(<<otherMethod>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|trait Main {
+       |  def main() = {
+       |    def method1(s : String => Int) = 123
+       |    def otherMethod(arg0: String): Int = ???
+       |    method1(otherMethod)
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda-2",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : (String, Float) => Int) = 123
+       |    method1(<<otherMethod>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|trait Main {
+       |  def main() = {
+       |    def method1(s : (String, Float) => Int) = 123
+       |    def otherMethod(arg0: String, arg1: Float): Int = ???
+       |    method1(otherMethod)
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda-0",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : => Int) = 123
+       |    method1(<<otherMethod>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : => Int) = 123
+       |    def otherMethod: Int = ???
+       |    method1(otherMethod)
+       |  }
+       |}
+       |
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda-0-with-fn-arg",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : String => Int) = s("123")
+       |    method1(<<lol>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : String => Int) = s("123")
+       |    def lol(arg0: String): Int = ???
+       |    method1(lol)
+       |  }
+       |}
+       |
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda-generic",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    val list = List(1, 2, 3)
+       |    list.map(<<otherMethod>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|trait Main {
+       |  def main() = {
+       |    val list = List(1, 2, 3)
+       |    def otherMethod(arg0: Int) = ???
+       |    list.map(otherMethod)
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  checkEdit(
+    "lambda-generic-filter",
+    """|
+       |trait Main {
+       |  def main() = {
+       |    val list = List(1, 2, 3)
+       |    list.filter(<<otherMethod>>)
+       |  }
+       |}
+       |
+       |""".stripMargin,
+    """|trait Main {
+       |  def main() = {
+       |    val list = List(1, 2, 3)
+       |    def otherMethod(arg0: Int) = ???
```

Comment:
I can create an issue later for an improvement, but now I would focus on wrapping this up instead of adding more features.
By the way: should I create multiple issues? That seems better so that the contribution can be small and easy (to invite other inexperienced contributors), in the sense of a "good first issue" tag kind of thing in github.